airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/performancetest/AbstractSourceFillDbWithTestData.java

```diff
@@ -0,0 +1,120 @@
+   *
+   * @return configured test database
+   * @throws Exception - might throw any exception during initialization.
+   */
+  protected abstract Database setupDatabase(String dbName) throws Exception;
+
+  /**
+   * Get a create table template for a DB
+   *
+   * @return a create tabple template, ex. "CREATE TABLE test.%s(id INTEGER PRIMARY KEY, %s)"
+   */
+  protected abstract String getCreateTableTemplate();
+
+  /**
+   * Get a test field'stype that will be used in DB for table creation.
+   *
+   * @return a test's field type. Ex: varchar(8)
+   */
+  protected abstract String getTestFieldType();
+
+  /**
+   * Get a INSERT query template for a DB
+   *
+   * @return an INSERT into table query template, ex. "INSERT INTO test.%s (%s) VALUES %s"
+   */
+  protected abstract String getInsertQueryTemplate();
+
+  protected String prepareCreateTableQuery(final String dbSchemaName,
+                                           final int numberOfColumns,
+                                           final String currentTableName) {
+
+    StringJoiner sj = new StringJoiner(",");
+    for (int i = 0; i < numberOfColumns; i++) {
+      sj.add(String.format(" %s%s %s", getTestColumnName(), i, getTestFieldType()));
+    }
+
+    return String.format(getCreateTableTemplate(), dbSchemaName, currentTableName, sj.toString());
+  }
+
+  // ex. INSERT INTO test.test_1 (id, test_column0, test_column1) VALUES (101,"zzz0", "sss0"), ("102",
+  // "zzzz1", "sss1");
+  protected String prepareInsertQueryTemplate(final String dbSchemaName,
+                                              final int numberOfColumns,
+                                              final int recordsNumber) {
+
+    StringJoiner fieldsNames = new StringJoiner(",");
+    fieldsNames.add("id");
+
+    StringJoiner baseInsertQuery = new StringJoiner(",");
+    baseInsertQuery.add("%s");
+
+    for (int i = 0; i < numberOfColumns; i++) {
+      fieldsNames.add(getTestColumnName() + i);
+      baseInsertQuery.add(String.format(TEST_VALUE_TEMPLATE, i));
+    }
+
+    StringJoiner insertGroupValuesJoiner = new StringJoiner(",");
+
+    for (int currentRecordNumber = 0; currentRecordNumber < recordsNumber; currentRecordNumber++) {
+      insertGroupValuesJoiner
+          .add("(" + String.format(baseInsertQuery.toString(), currentRecordNumber) + ")");
+    }
+
+    return String
+        .format(getInsertQueryTemplate(), dbSchemaName, "%s", fieldsNames.toString(),
+            insertGroupValuesJoiner.toString());
+  }
+
+  // ex. INSERT INTO "test100tables100recordsDb".test_0 (id,test_column0)
+  // VALUES (0,'Value t0'), (1,'Value t1');
+  protected String prepareInsertQueryTemplatePostgres(final String dbSchemaName,
+                                                      final int batchNumber,
+                                                      final int numberOfColumns,
+                                                      final int recordsNumber) {
```

Comment:
This method is used not only by Postgres, but by MSSQL as well. So its name can be more generic.

Reply:
Updated, thanks!